src/redis-cli.c

```diff
@@ -1142,6 +1148,62 @@ static sds cliFormatReplyCSV(redisReply *r) {
     return out;
 }
 
+static sds cliFormatReplyJson(sds out, redisReply *r) {
+    unsigned int i;
+
+    switch (r->type) {
+    case REDIS_REPLY_ERROR:
+        out = sdscat(out,"error:");
+        out = sdscatrepr(out,r->str,strlen(r->str));
+        break;
+    case REDIS_REPLY_STATUS:
+        out = sdscatrepr(out,r->str,r->len);
+        break;
+    case REDIS_REPLY_INTEGER:
+        out = sdscatprintf(out,"%lld",r->integer);
+        break;
+    case REDIS_REPLY_DOUBLE:
+        out = sdscatprintf(out,"%s",r->str);
+        break;
+    case REDIS_REPLY_STRING:
+    case REDIS_REPLY_VERB:
+        out = sdscatrepr(out,r->str,r->len);
+        break;
+    case REDIS_REPLY_NIL:
+        out = sdscat(out,"null");
+        break;
+    case REDIS_REPLY_BOOL:
+        out = sdscat(out,r->integer ? "true" : "false");
+        break;
+    case REDIS_REPLY_ARRAY:
+    case REDIS_REPLY_SET:
+    case REDIS_REPLY_PUSH:
+        out = sdscat(out,"[");
+        for (i = 0; i < r->elements; i++ ) {
+            out = cliFormatReplyJson(out, r->element[i]);
+            if (i != r->elements-1) out = sdscat(out,",");
+        }
+        out = sdscat(out,"]");
+        break;
+    case REDIS_REPLY_MAP:
+        out = sdscat(out,"{");
+        for (i = 0; i < r->elements; i += 2) {
+            out = cliFormatReplyJson(out, r->element[i]);
```

Comment:
Following up on @oranagra 's comment, and checking the JSON spec:
https://datatracker.ietf.org/doc/html/draft-crockford-jsonorg-json-04#section-1
we see:

> An object is an unordered collection of zero or more name/value
   pairs, where a name is a string, and a value is a string, number,
   boolean, null, object, or array.
   
   meaning on line 1191 we need to ensure that the name on the name/value pair is a string and not any other type.